src/components/buttons/ViewBtn.js

```diff
@@ -1,13 +1,14 @@
 import { EyeIcon } from '@heroicons/react/solid';
 import Link from 'next/link';
 
-export default function ViewBtn ({ id }) {
+export default function ViewBtn ({ id, onClick }) {
   return (
     <Link href={{ pathname: `/course/${id}` }}>
       <button
         id={'view-course-button-' + id}
         className='flex justify-center items-center gap-2 text-blue-400 rounded-full hover:shadow-md bg-blue-50 hover:bg-blue-400 hover:text-white p-1.5 transform transition-all duration-150 ease-in-out border-blue-400 border-2 focus:ring-2 ring-blue-400 outline-none'
         title='view course'
+        onClick={onClick}
```

Comment:
the onClick here is being passed in from another location. To avoid prop drilling into oblivion can we move the logic of the on click into the view button itself that way we can use it across the app without having to re do the function every time?

Reply:
Addressed